pkg/cloudprovider/cloudapi/internal/accounts_common.go

```diff
@@ -117,66 +109,37 @@ func (c *cloudCommon) updateCloudAccountConfig(client client.Client, credentials
 		return err
 	}
 
-	serviceConfigs, err := cloudServicesCreateFunc(currentConfig.namespacedName, cloudConvertedNewCredential, c.cloudSpecificHelper)
+	serviceConfig, err := cloudServicesCreateFunc(currentConfig.namespacedName, cloudConvertedNewCredential, c.cloudSpecificHelper)
 	if err != nil {
 		return err
 	}
-	serviceConfigMap := make(map[CloudServiceName]CloudServiceInterface)
-	for _, serviceCfg := range serviceConfigs {
-		serviceConfigMap[serviceCfg.GetName()] = serviceCfg
-	}
-
-	return currentConfig.update(serviceConfigMap, c.logger())
-}
-
-func (accCfg *cloudAccountConfig) update(newSvcConfigMap map[CloudServiceName]CloudServiceInterface, logger logging.Logger) error {
-	accCfg.mutex.Lock()
-	defer accCfg.mutex.Unlock()
-
-	existingSvcConfigMap := accCfg.serviceConfigs
-	for name, svcConfig := range existingSvcConfigMap {
-		newSvcCfg, found := newSvcConfigMap[name]
-		if !found {
-			// should not happen
-			continue
-		}
-		if err := svcConfig.updateServiceConfig(newSvcCfg); err != nil {
-			return err
-		}
-		logger.Info("Service config updated (api-clients to use new creds)", "account", accCfg.namespacedName,
```

Comment:
Do we have  log message to know if a service config is updated or not?

Reply:
should be ok